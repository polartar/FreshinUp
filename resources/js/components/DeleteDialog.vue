<template>
  <v-dialog
    v-bind="$attrs"
    max-width="500"
    v-on="$listeners"
  >
    <simple-confirm
      :class="{ deleting: isLoading }"
      :title="title"
      ok-label="Yes"
      cancel-label="No"
      @ok="$emit('confirm')"
      @cancel="$emit('cancel')"
    >
      <div class="py-5 px-2">
        <template v-if="isLoading">
          <div class="text-xs-center">
            <p class="subheading">
              Processing, please wait...
            </p>
            <v-progress-circular
              :rotate="-90"
              :size="200"
              :width="15"
              :value="progress"
              color="primary"
            >
              {{ progressStatus }}
            </v-progress-circular>
          </div>
        </template>
        <template v-else>
          <p class="subheading">
            item(s): {{ message }}
          </p>
        </template>
      </div>
    </simple-confirm>
  </v-dialog>
</template>

<script>
import simpleConfirm from 'fresh-bus/components/SimpleConfirm.vue'

export default {
  components: {
    simpleConfirm
  },
  props: {
    itemTitleProp: { type: String, default: 'title' },
    isLoading: { type: Boolean, default: false },
    progress: { type: Number, default: 0 },
    title: {
      type: String,
      default: 'Are you sure you want to delete the selected item(s)'
    },
    items: { type: Array, default: () => [] }
  },
  computed: {
    progressStatus () {
      return `${Math.round(this.progress * 100) / 100} %`
    },
    message () {
      return this.items.map(item => item[this.itemTitleProp]).join(', ')
    }
  }
}
</script>
