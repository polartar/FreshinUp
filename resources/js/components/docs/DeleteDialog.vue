<template>
  <v-dialog
    v-bind="$attrs"
    max-width="500"
    v-on="$listeners"
  >
    <simple-confirm
      :class="{ deleting: processing }"
      :title="dialogTitle"
      ok-label="Yes"
      cancel-label="No"
      @ok="onSubmitDelete"
      @cancel="onCancelDelete"
    >
      <div class="py-5 px-2">
        <template v-if="processing">
          <div class="text-xs-center">
            <p class="subheading">
              Processing, please wait...
            </p>
            <v-progress-circular
              :rotate="-90"
              :size="200"
              :width="15"
              :value="deletablesProgress"
              color="primary"
            >
              {{ deletablesStatus }}
            </v-progress-circular>
          </div>
        </template>
        <template v-else>
          <p class="subheading">
            <span v-if="deletables.length < 2">Document</span>
            <span v-else>Documents</span>
            : {{ deleteTemp | formatDeleteTitles }}
          </p>
        </template>
      </div>
    </simple-confirm>
  </v-dialog>
</template>

<script>
// TODO: replace with the one under components/DeleteDialog
import { deletables } from 'fresh-bus/components/mixins/Deletables'
import simpleConfirm from 'fresh-bus/components/SimpleConfirm.vue'
export default {
  components: {
    simpleConfirm
  },
  filters: {
    formatDeleteTitles (value) {
      return value.map(item => item.title).join(', ')
    }
  },
  mixins: [deletables],
  props: {
    processing: {
      type: Boolean,
      default: false
    },
    dialogTitle: {
      type: String,
      default: 'Are you sure you want to delete the selected item(s)'
    },
    deleteTemp: {
      type: Array,
      default: () => []
    },
    onSubmitDelete: {
      type: Function,
      required: true
    },
    onCancelDelete: {
      type: Function,
      required: true
    }
  }
}
</script>
