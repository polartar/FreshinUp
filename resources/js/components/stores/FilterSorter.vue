<template>
  <search-filter-sorter
    ref="filter"
    class="filter-transparent"
    expanded
    without-filter-label
    sort-label="Sort by"
    :sort-options="sortables"
    color="transparent"
    v-bind="$attrs"
    v-on="$listeners"
    @run="run"
    @clear="clearFilters"
  >
    <template v-slot:expanded="slotProps">
      <v-card-text class="px-0 pb-0">
        <v-layout
          row
          justify-space-between
        >
          <v-flex>
            <v-layout
              row
              justify-space-between
              align-center
              mb-2
            >
              <filter-label>
                Statuses
              </filter-label>
              <clear-button
                v-if="filters.status && filters.status.length > 0"
                color="white"
                @clear="filters.status = null;"
              />
            </v-layout>
            <multi-select
              v-model="filters.status"
              placeholder="Select Status"
              :items="statuses"
              item-value="id"
              item-text="name"
              select-all-name="All Status"
              solo
              flat
              hide-details
            />
          </v-flex>
          <v-flex
            ml-2
          >
            <v-layout
              row
              justify-space-between
              mb-2
            >
              <filter-label>
                Supplier
              </filter-label>
              <clear-button
                v-if="filters.supplier_uuid && filters.supplier_uuid.length > 0"
                color="white"
                @clear="filters.supplier_uuid = null; $refs.host.resetTerm()"
              />
            </v-layout>
            <multi-simple
              ref="supplier"
              v-model="filters.supplier_uuid"
              url="companies?filter[type_key]=supplier"
              term-param="filter[name]"
              results-id-key="uuid"
              placeholder="Select"
              background-color="white"
              class="mt-0 pt-0"
              height="48"
              not-clearable
              solo
              flat
            />
          </v-flex>
          <v-flex
            ml-2
          >
            <v-layout
              row
              justify-space-between
              mb-2
            >
              <filter-label>
                Hometown
              </filter-label>
              <clear-button
                v-if="filters.location_uuid"
                color="white"
                @clear="filters.location_uuid = null; $refs.location.resetTerm()"
              />
            </v-layout>
            <simple
              ref="location"
              url="foodfleet/locations"
              term-param="filter[name]"
              results-id-key="uuid"
              :value="filters.location_uuid"
              placeholder="Select"
              background-color="white"
              class="mt-0 pt-0"
              height="48"
              not-clearable
              solo
              flat
              @input="selectLocation"
            />
          </v-flex>
          <v-flex
            ml-2
          >
            <v-layout
              row
              justify-space-between
              mb-2
            >
              <filter-label>
                Tags
              </filter-label>
              <clear-button
                v-if="filters.store_tag_uuid && filters.store_tag_uuid.length > 0"
                color="white"
                @clear="filters.store_tag_uuid = null; $refs.tag.resetTerm()"
              />
            </v-layout>
            <multi-simple
              ref="tag"
              v-model="filters.store_tag_uuid"
              url="foodfleet/store-tags"
              term-param="filter[name]"
              results-id-key="uuid"
              placeholder="Search Tag"
              background-color="white"
              class="mt-0 pt-0"
              height="48"
              not-clearable
              solo
              flat
            />
          </v-flex>
        </v-layout>
      </v-card-text>
    </template>
  </search-filter-sorter>
</template>
<script>
import ClearButton from '~/components/ClearButton'
import FilterLabel from '~/components/FilterLabel'
import SearchFilterSorter from 'fresh-bus/components/search/filter-sorter'
import Simple from 'fresh-bus/components/search/simple'
import MultiSelect from 'fresh-bus/components/ui/FMultiSelect'
import MultiSimple from 'fresh-bus/components/ui/FMultiSimple'
export default {
  components: {
    Simple,
    MultiSimple,
    FilterLabel,
    ClearButton,
    MultiSelect,
    SearchFilterSorter
  },
  props: {
    filters: {
      type: Object,
      default: () => ({
        status: null,
        store_tag_uuid: null,
        location_uuid: null,
        supplier_uuid: null
      })
    },
    statuses: {
      type: Array,
      default: () => []
    },
    sortables: {
      type: Array,
      default: () => []
    }
  },
  watch: {
    filters: {
      handler: function (value) {
        const params = this.$refs.filter.getRunParams()
        this.run(params)
      },
      deep: true
    }
  },
  methods: {
    run (params) {
      let finalParams = {
        name: params.term,
        ...this.filters
      }
      if (this.filters.store_tag_uuid) {
        finalParams.store_tag_uuid = this.filters.store_tag_uuid.map(item => item.uuid)
      }
      if (this.filters.supplier_uuid) {
        finalParams.supplier_uuid = this.filters.supplier_uuid.map(item => item.uuid)
      }
      this.$emit('runFilter', finalParams)
    },
    selectLocation (location) {
      this.filters.location_uuid = location ? location.uuid : null
    },
    clearFilters (params) {
      this.$refs.tag.resetTerm()
      this.$refs.supplier.resetTerm()
      this.$refs.location.resetTerm()
      this.filters.status = this.filters.store_tag_uuid = this.filters.location_uuid = this.filters.supplier_uuid = null
    }
  }
}
</script>
<style lang="styl" scoped>
  /deep/ .data-time-picker.no-border input.field-input{
    border: none !important;
  }
  /deep/ .filter-sorter-expanded-layout{
    align-items: flex-end;
  }
  /deep/ .filter-sorter-expanded-layout>.flex.text-no-wrap>.v-btn{
    margin: 0;
    height: 48px;
  }
  .filter-transparent{
    box-shadow: none;
  }
</style>
