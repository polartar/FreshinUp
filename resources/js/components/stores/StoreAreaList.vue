<template>
  <div>
    <v-data-table
      v-model="selected"
      class="elevation-1"
      :headers="headers"
      :items="storeAreas"
      :rows-per-page-items="[5, 10, 15, 25, 30, 50]"
      :pagination.sync="pagination"
      :loading="isLoading"
      :total-items="totalItems"
      item-key="uuid"
      select-all
      disable-initial-sort
    >
      <v-progress-linear
        slot="progress"
        indeterminate
        height="10"
      />
      <template
        slot="headerCell"
        slot-scope="props"
      >
        <span>
          {{ props.header.text }}
        </span>
      </template>
      <template slot="no-data">
        <v-alert
          :value="true"
          color="error"
          icon="warning"
        >
          Sorry, nothing to display here :(
        </v-alert>
      </template>

      <template
        slot="items"
        slot-scope="props"
      >
        <td>
          <v-checkbox
            v-model="props.selected"
            primary
            hide-details
          />
        </td>
        <td>
          <div class="subheading primary--text">
            {{ props.item.name }}
          </div>
        </td>
        <td>
          <div class="grey--text">
            {{ props.item.state }}
          </div>
        </td>
        <td>
          <div class="grey--text">
            {{ props.item.radius }}
          </div>
        </td>
        <td class="justify-center">
          <v-btn
            large
            class="primary"
            @click="delete(props.item)"
          >
            Delete
          </v-btn>
        </td>
      </template>
    </v-data-table>
  </div>
</template>

<script>
import Pagination from 'fresh-bus/components/mixins/Pagination'
export const HEADERS = [
  { text: 'Area Name', sortable: true, value: 'name' },
  { text: 'State', sortable: true, value: 'state' },
  { text: 'Radius', sortable: true, value: 'radius' },
  { text: 'Manage', sortable: false, value: 'manage' }
]
export default {
  mixins: [
    Pagination
  ],
  props: {
    storeAreas: {
      type: Array,
      default: () => []
    }
  },
  data () {
    return {
      selected: [],
      headers: HEADERS,
      actionBtnTitle: 'Delete'
    }
  },
  methods: {
    delete (item) {
      this.$emit('manage', item.action)
    }
  }
}
</script>
