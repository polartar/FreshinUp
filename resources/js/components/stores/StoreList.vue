<template>
  <div>
    <v-data-table
      v-model="selected"
      class="elevation-1"
      :headers="headers"
      :items="stores"
      :rows-per-page-items="[5, 10, 15, 25, 30, 50]"
      :pagination.sync="pagination"
      :loading="isLoading"
      :total-items="totalItems"
      item-key="uuid"
      select-all
      disable-initial-sort
    >
      <v-progress-linear
        slot="progress"
        indeterminate
        height="10"
      />
      <template
        slot="headerCell"
        slot-scope="props"
      >
        <span v-if="selected.length > 1 && props.header.value === 'manage'">
          <v-menu offset-y>
            <v-btn
              slot="activator"
              color="primary"
              dark
            >
              Manage Multiple
            </v-btn>
            <v-list>
              <v-list-tile
                v-for="(item, index) in selectedStoreActions"
                :key="index"
                @click="manageMultiple(item.action)"
              >
                <v-list-tile-title>
                  {{ item.text }}
                </v-list-tile-title>
              </v-list-tile>
            </v-list>
          </v-menu>
        </span>
        <span v-else-if="selected.length > 1 && props.header.value === 'status'">
          <v-menu offset-y>
            <v-btn
              slot="activator"
              light
            >
              Change Statuses
            </v-btn>
            <v-list>
              <v-list-tile
                v-for="(item, index) in statuses"
                :key="index"
                @click="changeStatusMultiple(item.id)"
              >
                <v-list-tile-title>
                  {{ item.name }}
                </v-list-tile-title>
              </v-list-tile>
            </v-list>
          </v-menu>
        </span>

        <span v-else>
          {{ props.header.text }}
        </span>
      </template>
      <template slot="no-data">
        <v-alert
          :value="true"
          color="error"
          icon="warning"
        >
          Sorry, nothing to display here :(
        </v-alert>
      </template>

      <template
        slot="items"
        slot-scope="props"
      >
        <td>
          <v-checkbox
            v-model="props.selected"
            primary
            hide-details
          />
        </td>
        <td class="justify-center text-xs-center select-td">
          <v-select
            :items="statuses"
            :value="props.item.status"
            item-text="name"
            item-value="id"
            menu-props="auto"
            label="Status"
            hide-details
            single-line
            solo
            @change="changeStatus($event, props.item)"
          />
        </td>
        <td>
          <div class="subheading primary--text">
            {{ props.item.name }}
          </div>
          <div class="grey--text">
            {{ props.item.type && props.item.type.name }}
          </div>
        </td>
        <td>
          <v-chip
            v-for="(tag, index) in props.item.tags"
            :key="index"
          >
            {{ tag.name }}
          </v-chip>
        </td>
        <td class="text-xs-center">
          <div class="grey--text">
            {{ props.item.owner && props.item.owner.name }}
          </div>
          <div class="grey--text">
            {{ props.item.owner && props.item.owner.company_name }}
          </div>
        </td>
        <td>
          {{ props.item.state_of_incorporation }}
        </td>
        <td class="justify-center text-xs-center">
          <f-btn-menu
            :items="itemActions"
            item-label="text"
            @item="manage($event, props.item)"
          >
            Manage
          </f-btn-menu>
        </td>
      </template>
    </v-data-table>
  </div>
</template>

<script>
import Pagination from 'fresh-bus/components/mixins/Pagination'
import FormatDate from 'fresh-bus/components/mixins/FormatDate'
import FBtnMenu from 'fresh-bus/components/ui/FBtnMenu'
export default {
  components: { FBtnMenu },
  mixins: [
    Pagination,
    FormatDate
  ],
  props: {
    stores: {
      type: Array,
      default: () => []
    },
    statuses: {
      type: Array,
      default: () => []
    }
  },
  data () {
    return {
      selected: [],
      headers: [
        { text: 'Status', sortable: true, value: 'status' },
        { text: 'Fleet Member Name / Type', sortable: true, value: 'name,type' },
        { text: 'Tags', sortable: true, value: 'tags' },
        { text: 'Owned By', sortable: true, value: 'owner' },
        { text: 'State Of Incorporation', sortable: true, value: 'state_of_incorporation' },
        { text: 'Manage', sortable: true, value: 'manage' }
      ],
      itemActions: [
        { action: 'view', text: 'View / Edit' },
        { action: 'delete', text: 'Delete' }
      ],
      actionBtnTitle: 'Manage'
    }
  },
  computed: {
    selectedStoreActions () {
      if (!this.selected.length) return []
      let actions = []
      actions.push({ action: 'delete', text: 'Delete' })
      return actions
    }
  },
  methods: {
    manage (item, store) {
      this.$emit('manage-' + item.action, store)
      this.$emit('manage', item.action, store)
    },
    manageMultiple (action) {
      this.$emit('manage-multiple-' + action, this.selected)
      this.$emit('manage-multiple', action, this.selected)
      this.selected = []
    },
    changeStatus (value, store) {
      this.$emit('change-status', value, store)
    },
    changeStatusMultiple (value) {
      this.$emit('change-status-multiple', value, this.selected)
    }
  }
}
</script>

<style scoped lang="scss">
  .fresh-bus-admin-user-list {
    &__joined_date {
      white-space: nowrap;
    }
  }
  .highlight {
    background: #ffa;
  }
  table.v-table tbody td.select-td{
    padding-top: 5px;
    padding-bottom: 5px;
  }
</style>
