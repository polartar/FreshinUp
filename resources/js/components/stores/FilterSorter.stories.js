import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import MockAdapter from 'axios-mock-adapter'
import axios from 'axios/index'

// Components
import FilterSorter from './FilterSorter.vue'

let statuses = [
  { value: 1, text: 'Draft' },
  { value: 2, text: 'Pending' },
  { value: 3, text: 'Revision' },
  { value: 4, text: 'Rejected' },
  { value: 5, text: 'Approved' },
  { value: 6, text: 'On hold' }
]

let sortables = [
  { value: '-created_at', text: 'Newest' },
  { value: 'created_at', text: 'Oldest' },
  { value: 'title', text: 'Title (A - Z)' },
  { value: '-title', text: 'Title (Z - A)' }
]

const mock = new MockAdapter(axios)
mock.onGet('foodfleet/store-tags').reply(200, {
  data: [
    { uuid: 1, name: 'Store Tag 1' },
    { uuid: 2, name: 'Store Tag 2' },
    { uuid: 3, name: 'Store Tag 3' },
    { uuid: 4, name: 'Store Tag 4' }
  ]
})
mock.onGet('foodfleet/locations').reply(200, {
  data: [
    { uuid: 1, name: 'Location 1' },
    { uuid: 2, name: 'Location 2' },
    { uuid: 3, name: 'Location 3' },
    { uuid: 4, name: 'Location 4' }
  ]
})

storiesOf('FoodFleet|fleet-member/FilterSorter', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('default', () => ({
    components: { FilterSorter },
    methods: {
      filterStores (params) {
        action('Run')(params)
      }
    },
    template: `
      <v-container>
        <filter-sorter
          @runFilter="filterStores"
        />
      </v-container>
    `
  }))
  .add('with statuses', () => ({
    components: { FilterSorter },
    data () {
      return {
        statuses: statuses
      }
    },
    methods: {
      filterStores (params) {
        action('Run')(params)
      }
    },
    template: `
      <v-container>
        <filter-sorter
          :statuses="statuses"
          @runFilter="filterStores"
        />
      </v-container>
    `
  }))
  .add('with sortables', () => ({
    components: { FilterSorter },
    data () {
      return {
        sortables: sortables
      }
    },
    methods: {
      filterStores (params) {
        action('Run')(params)
      }
    },
    template: `
      <v-container>
        <filter-sorter
          :sortables="sortables"
          @runFilter="filterStores"
        />
      </v-container>
    `
  }))
