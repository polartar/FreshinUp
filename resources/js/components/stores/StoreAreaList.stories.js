import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

// Components
import StoreAreaList from './StoreAreaList'
import {FIXTURE_STORE_AREAS} from '../../../../tests/Javascript/__data__/storeAreas'


storiesOf('FoodFleet|store/StoreAreaList', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('stores is empty', () => ({
    components: { StoreAreaList },
    data () {
      return {
        storeAreas: [],
        pagination: {
          page: 1,
          rowsPerPage: 10,
          totalItems: 5
        },
        sorting: {
          descending: false,
          sortBy: ''
        }
      }
    },
    template: `
      <store-area-list
        :store-areas="storeAreas"
        :rows-per-page="pagination.rowsPerPage"
        :page="pagination.page"
        :total-items="pagination.totalItems"
        :sort-by="sorting.sortBy"
        :descending="sorting.descending"
      />
    `
  }))
  .add('Stores is set', () => ({
    components: { StoreAreaList },
    data () {
      return {
        storeAreas: FIXTURE_STORE_AREAS,
        pagination: {
          page: 1,
          rowsPerPage: 10,
          totalItems: 5
        },
        sorting: {
          descending: false,
          sortBy: ''
        }
      }
    },
    methods: {
      del (params) {
        action('manage-delete')(params)
      }
    },
    template: `
      <store-area-list
        :store-areas="storeAreas"
        :rows-per-page="pagination.rowsPerPage"
        :page="pagination.page"
        :total-items="pagination.totalItems"
        :sort-by="sorting.sortBy"
        :descending="sorting.descending"
        @manage-delete="del"
      />
    `
  }))
