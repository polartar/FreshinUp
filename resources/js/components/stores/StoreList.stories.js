import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

// Components
import StoreList from './StoreList.vue'

let stores = [
  {
    uuid: 1,
    status: 1,
    name: 'sint1233',
    type: 'Food Truck',
    events: [
      { uuid: 1 },
      { uuid: 2 },
      { uuid: 3 }
    ],
    tags: [
      { uuid: 1, name: 'Asian' },
      { uuid: 2, name: 'Vietnamese' }
    ],
    addresses: [{ uuid: 1, city: 'Salvador', street: 'Rua' }],
    created_at: '2019-09-24T06:33:05.000000Z',
    updated_at: '2019-09-24T11:14:21.000000Z'
  },
  {
    uuid: 2,
    status: 2,
    name: 'sint1233',
    type: 'Food Truck',
    events: [
      { uuid: 1 },
      { uuid: 2 }
    ],
    tags: [
      { uuid: 1, name: 'Asian' },
      { uuid: 2, name: 'Vietnamese' },
      { uuid: 3, name: 'Asian Fusion' }
    ],
    addresses: [{ uuid: 1, city: 'Los Angeles', street: 'Road' }],
    created_at: '2019-09-24T06:33:05.000000Z',
    updated_at: '2019-09-24T11:14:21.000000Z'
  },
  {
    uuid: 3,
    status: 3,
    name: 'sint1233',
    type: 'Food Truck',
    events: [
      { uuid: 1 },
      { uuid: 2 },
      { uuid: 3 }
    ],
    tags: [
      { uuid: 1, name: 'Asian' },
      { uuid: 2, name: 'Vietnamese' },
      { uuid: 3, name: 'Asian Fusion' }
    ],
    addresses: [{ uuid: 1, city: 'San Jose', street: 'Street' }],
    created_at: '2019-09-24T06:33:05.000000Z',
    updated_at: '2019-09-24T11:14:21.000000Z'
  }
]

let statuses = [
  { id: 1, name: 'Draft' },
  { id: 2, name: 'Pending' },
  { id: 3, name: 'Confirmed' },
  { id: 4, name: 'Past' },
  { id: 5, name: 'Cancelled' }
]

storiesOf('FoodFleet|store/StoreList', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('stores is empty', () => ({
    components: { StoreList },
    data () {
      return {
        stores: [],
        statuses: statuses,
        pagination: {
          page: 1,
          rowsPerPage: 10,
          totalItems: 5
        },
        sorting: {
          descending: false,
          sortBy: ''
        }
      }
    },
    template: `
      <store-list
        :stores="stores"
        :statuses="statuses"
        :rows-per-page="pagination.rowsPerPage"
        :page="pagination.page"
        :total-items="pagination.totalItems"
        :sort-by="sorting.sortBy"
        :descending="sorting.descending"
      />
    `
  }))
  .add('Stores is set', () => ({
    components: { StoreList },
    data () {
      return {
        stores: stores,
        statuses: statuses,
        pagination: {
          page: 1,
          rowsPerPage: 10,
          totalItems: 5
        },
        sorting: {
          descending: false,
          sortBy: ''
        }
      }
    },
    methods: {
      edit (params) {
        action('manage-edit')(params)
      },
      del (params) {
        action('manage-delete')(params)
      },
      multipleDelete (params) {
        action('manage-multiple-delete')(params)
      },
      changeStatus (status, store) {
        action('change-status')(status, store)
      },
      changeStatusMultiple (status, stores) {
        action('change-status-multiple')(status, stores)
      }
    },
    template: `
      <store-list
        :stores="stores"
        :statuses="statuses"
        :rows-per-page="pagination.rowsPerPage"
        :page="pagination.page"
        :total-items="pagination.totalItems"
        :sort-by="sorting.sortBy"
        :descending="sorting.descending"
        @manage-edit="edit"
        @manage-delete="del"
        @manage-multiple-delete="multipleDelete"
        @change-status="changeStatus"
        @change-status-multiple="changeStatusMultiple"
      />
    `
  }))
