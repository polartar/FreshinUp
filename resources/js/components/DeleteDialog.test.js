import { mount, shallowMount } from '@vue/test-utils'
import * as Stories from './DeleteDialog.stories'
import Component from './DeleteDialog.vue'
import { FIXTURE_STORES } from '../../../tests/Javascript/__data__/stores'

describe('components/DeleteDialog', () => {
  describe('Snapshots', () => {
    test('Basic', async () => {
      const wrapper = mount(Stories.Basic())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Loading', async () => {
      const wrapper = mount(Stories.Loading())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Props & Computed', () => {
    test('isLoading', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.isLoading).toBe(false)

      wrapper.setProps({
        isLoading: true
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.isLoading).toBe(true)
    })
    test('itemTitleProp', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.itemTitleProp).toEqual('title')

      wrapper.setProps({
        itemTitleProp: 'name'
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.itemTitleProp).toEqual('name')
    })
    test('title', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.title).toEqual('Are you sure you want to delete the selected item(s)')

      wrapper.setProps({
        title: 'Trash them all ?'
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.title).toEqual('Trash them all ?')
    })
    test('items', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.items).toHaveLength(0)

      wrapper.setProps({
        items: FIXTURE_STORES // any array of items
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.items).toEqual(FIXTURE_STORES)
    })
    test('progress', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.progress).toEqual(0)

      wrapper.setProps({
        progress: 67
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.progress).toEqual(67)
    })
    test('progressStatus', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.progressStatus).toEqual('0 %')

      wrapper.setProps({
        progress: 67
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.progressStatus).toEqual('67 %')
    })
    test('message', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.message).toEqual('')

      wrapper.setProps({
        items: FIXTURE_STORES // any array of items
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.message).toEqual(FIXTURE_STORES.map(item => item.title).join(', '))
    })
  })
})
