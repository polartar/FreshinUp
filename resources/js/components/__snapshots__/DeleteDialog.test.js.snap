// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DeleteDialog Snapshots Basic 1`] = `
<div
  class="container"
>
  <div
    class="v-dialog__container"
    style="display: block;"
    value="true"
  >
    <div
      class="v-dialog__content v-dialog__content--active"
      style="z-index: 202;"
      tabindex="-1"
    >
      <div
        class="v-dialog v-dialog--active"
        style="max-width: 500px;"
      >
        <div
          class="simple-confirm pa-3 v-card v-sheet theme--light"
        >
          <div
            class="v-card__title px-0 justify-center"
          >
            <div
              class="title font-weight-bold"
            >
              
        All to trash ?
      
            </div>
          </div>
           
          <hr
            class="v-divider theme--light"
          />
           
          <div
            class="py-5 px-2"
          >
            <p
              class="subheading"
            >
              
          item(s): Demo Admin, Level 1 User, Level 2 User, Level 3 User, Level 4 User, Level 5 User, Level 6 User, Level 7 User, Level 8 User, Colleague 1 User
        
            </p>
          </div>
           
          <div
            class="v-card__actions px-8"
          >
            <button
              class="v-btn v-btn--block theme--light light"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
      No
    
              </div>
            </button>
             
            <div
              class="spacer"
            />
             
            <button
              class="v-btn v-btn--block theme--light primary"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
      Yes
    
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/DeleteDialog Snapshots Loading 1`] = `
<div
  class="container"
>
  <div
    class="v-dialog__container"
    style="display: block;"
    value="true"
  >
    <div
      class="v-dialog__content v-dialog__content--active"
      style="z-index: 202;"
      tabindex="-1"
    >
      <div
        class="v-dialog v-dialog--active"
        style="max-width: 500px;"
      >
        <div
          class="simple-confirm pa-3 v-card v-sheet theme--light deleting"
        >
          <div
            class="v-card__title px-0 justify-center"
          >
            <div
              class="title font-weight-bold"
            >
              
        Are you sure you want to delete the selected item(s)
      
            </div>
          </div>
           
          <hr
            class="v-divider theme--light"
          />
           
          <div
            class="py-5 px-2"
          >
            <div
              class="text-xs-center"
            >
              <p
                class="subheading"
              >
                
            Processing, please wait...
          
              </p>
               
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="0"
                class="v-progress-circular primary--text"
                role="progressbar"
                style="height: 200px; width: 200px;"
              >
                <svg
                  style="transform: rotate(-90deg);"
                  viewBox="21.62162162162162 21.62162162162162 43.24324324324324 43.24324324324324"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    class="v-progress-circular__underlay"
                    cx="43.24324324324324"
                    cy="43.24324324324324"
                    fill="transparent"
                    r="20"
                    stroke-dasharray="125.664"
                    stroke-dashoffset="0"
                    stroke-width="3.243243243243243"
                  />
                  <circle
                    class="v-progress-circular__overlay"
                    cx="43.24324324324324"
                    cy="43.24324324324324"
                    fill="transparent"
                    r="20"
                    stroke-dasharray="125.664"
                    stroke-dashoffset="125.66370614359172px"
                    stroke-width="3.243243243243243"
                  />
                </svg>
                <div
                  class="v-progress-circular__info"
                >
                  
            0 %
          
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="v-card__actions px-8"
          >
            <button
              class="v-btn v-btn--block theme--light light"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
      No
    
              </div>
            </button>
             
            <div
              class="spacer"
            />
             
            <button
              class="v-btn v-btn--block theme--light primary"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
      Yes
    
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
